github/commitstatus_test.go

```diff
@@ -78,22 +103,23 @@ func TestGitHubStatusSuccessMockAPI(t *testing.T) {
 					rateLimitReset:     now.Unix(),
 				},
 			},
+			wantSleeps: nil,
 		},
 		{
 			name: "Rate limited at the first attempt, success at the second attempt",
 			response: []mockedResponse{
 				{
 					status:             http.StatusForbidden,
 					rateLimitRemaining: emptyRateRemaining,
-					// Keep this value low to prevent tests to sleep for too long.
-					rateLimitReset: now.Add(1 * time.Second).Unix(),
+					rateLimitReset:     now.Add(42 * time.Second).Unix(),
 				},
 				{
 					status:             http.StatusCreated,
 					rateLimitRemaining: fullRateRemaining,
 					rateLimitReset:     now.Add(1 * time.Hour).Unix(),
 				},
 			},
+			wantSleeps: []time.Duration{42 * time.Second},
```

Comment:
What do you think about adding 1 more test, with jitter this time? I think jitter is implicitly set to 0 currently.

Reply:
I agree with you. We don't actually test that line 309 `sleepTime += jitter` is there.

Reply:
Jitter is covered in the unit tests for checkForRetry, see for example https://github.com/Pix4D/cogito/blob/master/github/commitstatus_private_test.go#L66. As such, I think this is OK as-is, unless I am missing something?

Reply:
> I think this is OK as-is, unless I am missing something.

I also think so from the testing side. From the clarity point of view, I am still wondering if its worth adding an explicit field for jitter when initializing `github.Target{}` to improve test clarity and avoid people asking questions like mine.

On the other hand, this is normal Go behavior; initialize to the 0 value of that type.

Reply:
> Jitter is covered in the unit tests for `checkForRetry`, see for example https://github.com/Pix4D/cogito/blob/master/github/commitstatus_private_test.go#L66. As such, I think this is OK as-is, unless I am missing something?

All good then. I only checked the test files modified in the PR. Sorry.

Reply:
> On the other hand, this is normal Go behavior; initialize to the 0 value of that type.

Exactly, this is why I do not touch these fields.

There are other cases where I do state the default, for example

```go
wantSleeps: nil,
```

is technically useless also. I make it explicit because I think we want to make it obvious that we expect 0 sleeps in that test case.

The jitter in my opinion is different: we simply do not care (in these tests).

At the end, it is subjective. I'd rather leave as-is for these reasons. If you are uneasy, I can make the jitter explicit too.

Reply:
> At the end, it is subjective. I'd rather leave as-is for these reasons. If you are uneasy, I can make the jitter explicit too.

No it's fine as it is. I was also unsure if it's required or not when I asked the question, so I wanted to double-check and see your opinion on it.